### ISSUE TITLE
package options in profile trigger rebuild, but not the command line

### ISSUE THREAD

**Issue by AlexandreBossard** [conan-io/conan#5385]
To help us debug your issue please explain:

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant.
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

conan 1.16.1

I want to add a package option in my profile (questionnable, I know):
```
## myprofile-release
## ...
[options]
mypackage:myoption=True
```
But it appears that it triggers a rebuild of all the dependencies. `mypackage` is actually the consumer of all those deps.
If I type `conan install .. --profile myprofile-release --options mypackage:myoption=True` everything is fine.

**Comment by lasote** [comment]
> I want to add a package option in my profile (questionnable, I know):

No, that is totally fine.

I would need more information, that shouldn't happen, it should change the package_id of `mypackage` but not for the dependencies of `mypackage`.
Do you have some traces to try to guess something?

**Comment by AlexandreBossard** [comment]
TLDR: my bad, some of my profiles were not set up properly, but conan did not help :/
Here are my profiles for context:

```
$ cat ~/.conan/profiles/gcc8
[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=8
compiler.libcxx=libstdc++11

[env]
CONAN_CMAKE_GENERATOR=Ninja
CC=gcc-8
CXX=g++-8
```
```
$ cat ~/.conan/profiles/gcc8-release
include(gcc8)
[settings]
build_type=Release
```
```
$ cat ~/.conan/profiles/gcc8-release-shared
include(gcc8)
[options]
mypackage:myoption=True
```

As you can see, `gcc8-release-shared` includes `gcc8` and not `gcc8-release`. Thus, not setting the build-type.

How a build_type=None is valid ? Also the output does not print the used but not set variables that are used by the package_id.

```
$  conan install .. --profile gcc8-release-shared
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=8
os=Linux
os_build=Linux
[options]
*:shared=False
tanker:tankerlib_shared=True
[build_requires]
[env]
CC=gcc-8
CONAN_CHANNEL=stable
CONAN_CMAKE_GENERATOR=Ninja
CONAN_USERNAME=tanker
CXX=g++-8
```
```
$ conan install .. --profile gcc8-release --options mypackage:myoption=True
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=8
os=Linux
os_build=Linux
[options]
*:shared=False
tanker:tankerlib_shared=True
[build_requires]
[env]
CC=gcc-8
CONAN_CHANNEL=stable
CONAN_CMAKE_GENERATOR=Ninja
CONAN_USERNAME=tanker
CXX=g++-8

```

**Comment by lasote** [comment]
The `None`  value is by default allowed at the `settings.yml` file, you can remove the None value to always request a value. Honestly, I don't remember why we allow that but anyway it cannot be changed now due to the stability commitment.

`build_type: [None, Debug, Release, RelWithDebInfo, MinSizeRel]`

**Comment by AbrilRBS** [comment]
Closing. Profiles are always output when working with them in new Conan versions.

As for `None` being allowed, this helps to model situations like in https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html where the `BUILD_TYPE` can be an empty string, even if discouraged
